lib/ui/common/ActionSheetListItem.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:flutter/material.dart';
+
+enum ActionType { simple, withIcon, selectable }
+
+class ActionSheetListItemViewModel {
+  String title;
+  String icon;
+  ActionType type;
+  String checkBoxIcon;
+  bool isSelected;
+  bool isDestructive;
+  Function onTap;
+
+  ActionSheetListItemViewModel(
+      {this.title,
+      this.icon,
+      this.type,
+      this.checkBoxIcon,
+      this.isSelected,
+      this.isDestructive,
+      this.onTap});
+}
+
+class ActionSheetListItemStyle {
+  final Color actionItemBackgroundColor;
+
+  final TextStyle actionTextStyle;
+  final TextStyle destructiveTextStyle;
+
+  final EdgeInsets actionItemEdgePadding;
+
+  final double actionItemHeight;
+  final double iconLineSpace;
+  final double actionItemElevation;
+  final double iconHeight;
+
+  final int maxLines;
+
+  ActionSheetListItemStyle(
+      {this.actionItemBackgroundColor,
+      this.actionTextStyle,
+      this.destructiveTextStyle,
+      this.actionItemEdgePadding,
+      this.actionItemHeight,
+      this.iconLineSpace,
+      this.actionItemElevation,
+      this.iconHeight,
+      this.maxLines});
+
+  factory ActionSheetListItemStyle.defaultStyle() {
+    return ActionSheetListItemStyle(
+      actionItemBackgroundColor: const Color(0x00000000),
+      actionTextStyle: const TextStyle(
+          fontSize: 17,
+          fontWeight: FontWeight.normal,
+          color: Color(0xFF1a1b46)),
+      destructiveTextStyle: const TextStyle(
+          fontSize: 17,
+          fontWeight: FontWeight.normal,
+          color: Color(0xFFff6060)),
+      actionItemEdgePadding: const EdgeInsets.symmetric(horizontal: 32),
+      actionItemHeight: 70,
+      iconLineSpace: 21.5,
+      actionItemElevation: 0,
+      iconHeight: 20,
+      maxLines: 1,
+    );
+  }
+
+  factory ActionSheetListItemStyle.selectableStyle() {
```

Comment:
Add a TODO comment to refactor this on another PR style refactor. This should be outside of this class should only have the default one